homeassistant/components/niko_home_control/climate.py

```diff
@@ -0,0 +1,100 @@
+"""Support for Niko Home Control thermostats."""
+
+from typing import Any
+
+from nhc.const import THERMOSTAT_MODES
+from nhc.thermostat import NHCThermostat
+
+from homeassistant.components.climate import (
+    PRESET_ECO,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.components.sensor import UnitOfTemperature
+from homeassistant.const import ATTR_TEMPERATURE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import NikoHomeControlConfigEntry
+from .entity import NikoHomeControlEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: NikoHomeControlConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Niko Home Control thermostat entry."""
+    controller = entry.runtime_data
+
+    async_add_entities(
+        NikoHomeControlClimate(thermostat, controller, entry.entry_id)
+        for thermostat in controller.thermostats.values()
+    )
+
+
+class NikoHomeControlClimate(NikoHomeControlEntity, ClimateEntity):
+    """Representation of a Niko Home Control thermostat."""
+
+    _attr_supported_features: ClimateEntityFeature = (
+        ClimateEntityFeature.PRESET_MODE | ClimateEntityFeature.TARGET_TEMPERATURE
+    )
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_name = None
+    _action: NHCThermostat
+
+    _attr_translation_key = "nhc_thermostat"
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available hvac modes."""
+        return [HVACMode.OFF, HVACMode.COOL, HVACMode.AUTO]
+
+    @property
+    def preset_modes(self):
+        """Return the list of available preset modes."""
+        return [
+            "day",
+            "night",
+            PRESET_ECO,
+            "prog1",
+            "prog2",
+            "prog3",
+        ]
```

Comment:
What about reusing other existing presets here as well and map them to the nbc values like `night` == `PRESET_SLEEP` and `day` == `PRESET_HOME`.

Reply:
Not all modes that are mentioned here have an icon assigned.